uitools/cpp/Esri/ArcGISRuntime/Toolkit/SearchViewController.cpp

```diff
@@ -423,13 +463,28 @@ namespace Toolkit {
 
       addGeoElementToOverlay(m_graphicsOverlay, m_selectedResult->geoElement());
 
+      auto connection = std::make_shared<QMetaObject::Connection>();
       if (auto sceneView = qobject_cast<SceneViewToolkit*>(m_geoView))
       {
+        // When the geoview changes, update the lastsearcharea
+        *connection = connect(sceneView, &SceneViewToolkit::viewpointChanged, this, [sceneView, this, connection]()
+                              {
+                                disconnect(*connection);
+                                auto extent = sceneView->currentViewpoint(ViewpointType::BoundingGeometry).targetGeometry().extent();
+                                setLastSearchArea(extent);
+                              });
```

Comment:
💭 🔧  As I understand it - these connections are here because we need to find the difference between the currently progmatically "set" viewpoint, and user navigation away from that set viewpoint.

I think there's some room for simplification here. 

In the `setViewpoint` lambda at line 161, you could test when  `isNavigating() == false`, and then update `m_lastSearchArea`. This means `m_lastSearchArea` is updated to whatever the viewpoint is changed ONLY when the user is NOT navigating.

If this works, it would remove the 4 singleshot connections that are currently used.

Reply:
Using your idea to only set the lastsearcharea after the viewpoint has changed (waiting for geoview signal) and that the geoview->isnotnavigating.
It does seem working, but after some debugging I think there might be some cases in which it will not work. For example by setting a qdebug() every time that the lastsearcharea is set, I can see that it is set after a search, suggestion selected (viewpoint moved), but also when the user pans or zooms. If the user will do a small pan, the lastsearcharea is set, invalidating the viewpoint that should only be set AFTER a search, not a pan or zoom. demo showing it:
https://user-images.githubusercontent.com/91129623/146052257-f49e7bdd-5865-4296-bdff-12715a852ec1.mp4

Reply:
Good investigation. I can also see this bug. The zoom behaviour seems particularly egregious. This is not how I'd expect the API to behave. Let's open a ticket in the Qt repo, we might also be able to bugfix ourselves also.

Reply:
🔧 

Overall it sounds like we need an API improvement in this area. As written `isNavigating` gives too many false positives.

Your initial solution is good enough, but what I would like to see is some boilerplate removed.

Within the `internal` folder can you make a new C++ header file, (something like `singleShotConnection.h`)

You can make a helper for singleshot by doing something like this:

```c++
namespace Esri {
namespace ArcGISRuntime {
namespace Toolkit {

template <typename Sender, typename Signal, typename Receiver, typename Slot>
QMetaObject::Connection singleShotConnection(Sender* sender, Signal&& signal, Receiver* reciever, Slot&& slot)
{
  auto connection = std::make_shared<QMetaObject::Connection>();
  *connection = QObject::connect(sender, std::forward<Signal>(signal),
                                 reciever, [connection, slot = std::forward<Slot>(slot)](auto&&...args)
                                 {
                                   QObject::disconnect(*connection);
                                   slot(args...);
                                 });
  return *connection;
}
```

And update all single-shot connections in the codebase. E.g.:

```cpp
singleShotConnection(sceneView, &SceneViewToolkit::viewpointChanged, this, [sceneView, this]()
{
  auto extent = sceneView->currentViewpoint(ViewpointType::BoundingGeometry).targetGeometry().extent();
  setLastSearchArea(extent);
});
```

I think that will go a long way towards maintainability and readability.